appendix/stata-guide.tex

```diff
@@ -177,127 +216,204 @@ \subsection{Writing loops}
+making it 8 spaces more indented than the main code.
+In some code editors this indentation can be achieved by using the tab button.
+However, the type of tab used in the Stata do-file editor does not always display the same across platforms,
+such as when publishing the code on GitHub.
+Therefore we recommend that indentation be 4 manual spaces instead of a tab.
 
 \codeexample{stata-whitespace-indentation.do}{./code/stata-whitespace-indentation.do}
 
 \subsection{Writing conditional expressions}
 
 All conditional (true/false) expressions should be within at least one set of parentheses.
 The negation of logical expressions should use bang (\texttt{!}) and not tilde (\texttt{\~}).
+Always use explicit truth checks (\texttt{if `value'==1}) rather than implicits (\texttt{if `value'}).
+Always use the \texttt{missing(`var')} function instead of arguments like (\texttt{if `var'<=.}),
+and always consider whether missing values will affect the evaluation conditional expressions.
 
 \codeexample{stata-conditional-expressions1.do}{./code/stata-conditional-expressions1.do}
 
-You should also always use \texttt{if-else} statements when applicable even if you can express the same
-thing with two separate \texttt{if} statements. When using \texttt{if-else} statements you are
-communicating to anyone reading your code that the two cases are mutually exclusive in an \texttt{if-else} statement
-which makes your code more readable. It is also less error-prone and easier to update.
+Use \texttt{if-else} statements when applicable
+even if you can express the same thing with two separate \texttt{if} statements.
+When using \texttt{if-else} statements you are communicating to anyone reading your code
+that the two cases are mutually exclusive which makes your code more readable.
+It is also less error-prone and easier to update if you want to change the condition.
 
 \codeexample{stata-conditional-expressions2.do}{./code/stata-conditional-expressions2.do}
 
 \subsection{Using macros}
 
-Stata has several types of \textbf{macros} where numbers or text can be stored temporarily, but the two most common
-macros are \textbf{local} and \textbf{global}. Locals should always be the default type and globals should only
-be used when the information stored is used in a different do-file. Globals are error-prone since they are
-active as long as Stata is open, which creates a risk that a global from one project is incorrectly used in
-another, so only use globals where they are necessary. Our recommendation is that globals should only be defined in the \textbf{master do-file}.
-All globals should be referenced using both the the dollar sign and the curly brackets around their name;
+Stata has several types of \textbf{macros} where numbers or text can be stored temporarily,
+but the two most common macros are \textbf{local} and \textbf{global}.
+All macros should be defined using the \texttt{=} operator.
```

Comment:
I think we should discuss this.

Reply:
Agreed that this should not be our recommendation. When the `=` operator is used long strings are truncated. So unless you know your strings are short then ```local mystr "`str1' `str2'"``` should always be preferred to ```local mystr = "`str1'" + "`str2'"```

Reply:
I didn't know that. At what length does this happen? I generally use:
```
local mystr = "`str1' `str2'"
```

Reply:
Moved to #380

Reply:
moved to : https://github.com/worldbank/d4di/issues/380